### ISSUE TITLE
TypeScript errors TS4023, TS4094

### ISSUE THREAD

**Issue by bobgubko** [RobinTail/express-zod-api#1182]
I recently upgraded to the latest epxress-zod-api version from 9.*, adjusted the codebase and at the end got a bunch of TS4023, TS4094 errors. I know I'm not the first one to open this issue and I know that you recommend to disable declarations in `tsconfig.json`:

> If facing Typescript errors TS4023 or TS4094, ensure disabling declaration option in your tsconfig.json.

However in my case declarations cannot be disabled. Is there a way to make it work with declarations enabled or my only option is to stay on 9.* version?

**Comment by RobinTail** [comment]
Could you please provide the complete text of the error messages, @bobgubko ?

**Comment by RobinTail** [comment]
> However in my case declarations cannot be disabled. Is there a way to make it work with declarations enabled or my only option is to stay on 9.* version?

No, I can fix it by exposing the types needed for you to keep the declarations, but I need to know which ones are missing.
For that I'd like to ask you to provide the complete error messages so I could help you, @bobgubko

**Comment by bobgubko** [comment]
Here are the errors I'm getting:

<details>
<summary>Errors</summary>

```
error TS4023: Exported variable '[..]' has or is using name 'BasicSecurity' from external module "[...]/node_modules/express-zod-api/dist/index" but cannot be named.
error TS4023: Exported variable '[..]' has or is using name 'BearerSecurity' from external module "[...]/node_modules/express-zod-api/dist/index" but cannot be named.
error TS4023: Exported variable '[..]' has or is using name 'CommonConfig' from external module "[...]/node_modules/express-zod-api/dist/index" but cannot be named.
error TS4023: Exported variable '[..]' has or is using name 'CookieSecurity' from external module "[...]/node_modules/express-zod-api/dist/index" but cannot be named.
error TS4023: Exported variable '[..]' has or is using name 'CustomHeaderSecurity' from external module "[...]/node_modules/express-zod-api/dist/index" but cannot be named.
error TS4023: Exported variable '[..]' has or is using name 'InputSecurity' from external module "[...]/node_modules/express-zod-api/dist/index" but cannot be named.
error TS4023: Exported variable '[..]' has or is using name 'MiddlewareDefinition' from external module "[...]/node_modules/express-zod-api/dist/index" but cannot be named.
error TS4023: Exported variable '[..]' has or is using name 'OAuth2Security' from external module "[...]/node_modules/express-zod-api/dist/index" but cannot be named.
error TS4023: Exported variable '[..]' has or is using name 'OpenIdSecurity' from external module "[...]/node_modules/express-zod-api/dist/index" but cannot be named.
error TS4023: Exported variable '[..]' has or is using name 'ResultHandlerDefinition' from external module "[...]/node_modules/express-zod-api/dist/index" but cannot be named.
error TS4094: Property 'descriptions' of exported class expression may not be private or protected.
error TS4094: Property 'handler' of exported class expression may not be private or protected.
error TS4094: Property 'methods' of exported class expression may not be private or protected.
error TS4094: Property 'middlewares' of exported class expression may not be private or protected.
error TS4094: Property 'mimeTypes' of exported class expression may not be private or protected.
error TS4094: Property 'resultHandler' of exported class expression may not be private or protected.
error TS4094: Property 'schemas' of exported class expression may not be private or protected.
error TS4094: Property 'scopes' of exported class expression may not be private or protected.
error TS4094: Property 'siblingMethods' of exported class expression may not be private or protected.
error TS4094: Property 'statusCodes' of exported class expression may not be private or protected.
error TS4094: Property 'tags' of exported class expression may not be private or protected.
```
</details>

Isn't it easier to simply export all types used in the library so there's no need to worry about those errors?

**Comment by RobinTail** [comment]
Thank you for elaborating, @bobgubko .

> Isn't it easier to simply export all types used in the library so there's no need to worry about those errors?

I'm trying to be careful with what I export, because making backwards incompatible changes to the exposed entities is a breaking change requiring next major version according to SemVer.

> TS4023

Ok. This is clear how to fix, exactly by exposing those types.

> TS4094

This is weird. Need to do a research.

**Comment by RobinTail** [comment]
What is your version of Typescript, @bobgubko ?

**Comment by RobinTail** [comment]
Dear @bobgubko ,

could you please try version `12.1.0-beta1` and tell me if it fixes the issue?

**Comment by bobgubko** [comment]
It did fix the issue! 0 errors during build.

**Comment by RobinTail** [comment]
Awesome, thank you for letting me know.
I'm going to release it then, @bobgubko

**Comment by alindsay55661** [comment]
I am getting the same error for a different type (`Metadata`):

```
error TS4023: Exported variable '[...]' has or is using name Metadata from external module '[...]node_modules/.pnpm/express-zod-api@14.0.0_@types+express@4.17.20_@types+http-errors@2.0.3_@types+node@20.8.7_exp_ul3mzpgqwyhnfja3iatztge55q/node_modules/express-zod-api/dist/index' but cannot be named.
```

**Comment by RobinTail** [comment]
You should have created an issue for that, @alindsay55661 .
I may have not noticed your comment.

**Comment by RobinTail** [comment]
Fixed in version 14.0.3, @alindsay55661
